nexus/tests/integration_tests/console_api.rs

```diff
@@ -467,21 +467,46 @@ async fn test_login_redirect_simple(cptestctx: &ControlPlaneTestContext) {
     // encoded path is /abc/def
     expect_redirect(
         testctx,
-        "/login?state=%2Fabc%2Fdef",
-        &format!("{}?state=%2Fabc%2Fdef", expected),
+        "/login?redirect_uri=%2Fabc%2Fdef",
+        &format!("{}?redirect_uri=%2Fabc%2Fdef", expected),
     )
     .await;
 
     // if state param comes in not URL encoded, we should still URL encode it
     expect_redirect(
         testctx,
-        "/login?state=/abc/def",
-        &format!("{}?state=%2Fabc%2Fdef", expected),
+        "/login?redirect_uri=/abc/def",
+        &format!("{}?redirect_uri=%2Fabc%2Fdef", expected),
     )
     .await;
+}
+
+// reject anything that's not a relative URI
+#[nexus_test]
+async fn test_bad_redirect_uri(cptestctx: &ControlPlaneTestContext) {
+    let testctx = &cptestctx.external_client;
 
-    // empty state param gets dropped
-    expect_redirect(testctx, "/login?state=", &expected).await;
+    let paths = [
+        "/login",
+        "/login/my-silo/local",
+        "/login/my-silo/saml/my-idp",
+        "/login/my-silo/saml/my-idp/redirect",
+    ];
+    let bad_uris = ["foo", "", "http://example.com"];
```

Comment:
Erroring out on `?redirect_uri=` feels a little un-generous when we could treat it as `None` instead, but I'm not sure how to accomplish this. There's probably a serde attr one-liner I could put on the param that would make it handle empty string specially.